File: spring-boot-backend/src/main/java/com/qa/springbootbackend/service/BuyerService.java
package com.qa.springbootbackend.service;

import java.util.List;
import java.util.Optional;

import com.qa.springbootbackend.model.Buyer;


public interface BuyerService {    
    public Optional<Buyer> getBuyer(Long id);
    public List<Buyer> getAllBuyers();
    public void deleteBuyer(Long id);
    public Buyer updateBuyer(Buyer buyer);
    public Buyer saveBuyer(Buyer buyer); 
}

File: react-frontend/src/service/BuyerService.js
import React from "react";
import axios from 'axios';

const getAllBuyersUrl = "http://localhost:8080/getAllBuyers";
const findBuyerUrl = "http://localhost:8080/getBuyer/";
const addBuyerUrl =  "http://localhost:8080/saveBuyer";
const updateBuyerUrl =  "http://localhost:8080/updateBuyer";
const deleteBuyerUrl =  "http://localhost:8080/deleteBuyer/";

class BuyerService{

    getAllBuyers(){
        return axios.get(getAllBuyersUrl);
    }

    findBuyer(id){
        return axios.get(findBuyerUrl+id);        
    }

    addBuyer(buyer){
        return axios.post(addBuyerUrl, buyer);
    }

    updateBuyer(buyer){
        return axios.put(updateBuyerUrl, buyer);
    }

    deleteBuyer(id){
        return axios.delete(deleteBuyerUrl+id);
    }

}
export default new BuyerService();